python/ray/serve/backend_state.py

```diff
@@ -596,22 +621,35 @@ def delete_backend(self, backend_tag: BackendTag,
         return new_goal_id
 
     def _stop_wrong_version_replicas(
-            self, backend_tag: BackendTag, version: str,
-            graceful_shutdown_timeout_s: float) -> int:
+            self, backend_tag: BackendTag, target_replicas: int,
+            target_version: str, graceful_shutdown_timeout_s: float) -> int:
         # NOTE(edoakes): this short-circuits when using the legacy
         # `create_backend` codepath -- it can be removed once we deprecate
         # that as the version should never be None.
-        if version is None:
+        if target_version is None:
             return
 
```

Comment:
do we have any check that target_replica cannot be 0?

Reply:
target_replicas can be 0, this happens in the case when we're deleting a backend. in that case, we just want to tear everything down anyways so it doesn't matter how many we stop here.